### ISSUE TITLE
Make jackson dependency conditional in spring-web (reactive/classic)

### ISSUE THREAD

**Issue by ia3andy** [quarkusio/quarkus#24888]
With this it won't be necessary to add jackson manually when using spring-web and it will be compatible with both resteasy-reactive and classic.

With this we can remove jackson from the spring-web codestart:
https://github.com/quarkusio/quarkus/blob/main/independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/extension-codestarts/spring-web-codestart/codestart.yml?rgh-link-date=2022-01-28T13%3A16%3A07Z#L19

_Originally posted by @ia3andy in https://github.com/quarkusio/quarkus/issues/23265#issuecomment-1096438592_

**Comment by geoand** [comment]
Actually, it sort of works like this already.

We have `quarkus-spring-web-resteasy-classic` which is included automatically if `quarkus-spring-web` and `quarkus-resteasy-jackson` are present.
Similarly for RESTEasy Reactive

**Comment by ia3andy** [comment]
@geoand but here we want to remove the need of having `quarkus-resteasy-jackson` when using `quarkus-spring-web`, ex:
- resteasy-reactive + spring-web => auto include resteasy-reactive-jackson
- resteasy + spring-web => auto include resteasy-jackson
- spring-web => auto include resteasy-reactive-jackson

**Comment by geoand** [comment]
Yeah, I get it, but given what we already do with conditional dependencies, it won't work

**Comment by geoand** [comment]
We do we need a solution to this? I don't see this being a real issue...

Any solution would likely break the way things are done.

**Comment by ia3andy** [comment]
This seems logical to me, I don't see why spring-web shouldn't work out of the box.

**Comment by aureamunoz** [comment]
@aureamunoz take a look
